src/main/kotlin/se/uulm/snowballr/backend/service/UserService.kt

```diff
@@ -155,4 +164,35 @@ class UserService(
 
         return tokens
     }
+
+    override suspend fun updateUser(request: UserOuterClass.User.Update): UserOuterClass.User {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
+
+        // Check that user to update exists in the database
+        val requestedUserId = parseUUID(request.user.id, EntityType.USER)
+        val requestedUser = userRepo.getUserById(requestedUserId)
+
+        // Check whether the current user is a server admin if the role is changed or the requested user is different
+        // from the current user
+        if (request.mask.pathsList.contains("role") || currentUser.id != requestedUser.id) {
+            verifyServerAdminRole(
+                currentUser,
+            ) {
+                UnauthorizedException.Single(
+                    EntityType.USER,
+                    requestedUser.id.toString(),
+                    AccessType.UPDATE,
+                    currentUser.id.toString(),
+                )
+            }
```

Comment:
You can use `it` for the last param.
```suggestion
            verifyServerAdminRole(currentUser) {
                UnauthorizedException.Single(EntityType.USER, requestedUser.id.toString(), AccessType.UPDATE, it)
            }
```

I'm not happy with the way we treat this now, but I don't see another solution, so we keep this.